crates/tx5-connection/src/conn.rs

```diff
@@ -0,0 +1,200 @@
+
+                while let Some(cmd) = cmd_recv.recv().await {
+                    match cmd {
+                        ConnCmd::SigRecv(sig) => {
+                            use tx5_signal::SignalMessage::*;
+                            match sig {
+                                HandshakeReq(oth_nonce) => {
+                                    client
+                                        .send_handshake_res(
+                                            &pub_key2, oth_nonce,
+                                        )
+                                        .await?;
+                                    sent_our_res = true;
+                                }
+                                HandshakeRes(res_nonce) => {
+                                    if res_nonce != nonce {
+                                        return Err(Error::other(
+                                            "nonce mismatch",
+                                        ));
+                                    }
+                                    got_peer_res = true;
+                                }
+                                _ => {
+                                    return Err(Error::other(
+                                        "invalid message during handshake",
+                                    ));
+                                }
+                            }
+                        }
+                        ConnCmd::Close => {
+                            return Err(Error::other("close during handshake"))
+                        }
+                    }
+                    if got_peer_res && sent_our_res {
+                        break;
+                    }
+                }
+
+                Result::Ok(())
+            };
+
+            match tokio::time::timeout(config.max_idle, handshake_fut).await {
+                Err(_) | Ok(Err(_)) => {
+                    client.close_peer(&pub_key2).await;
+                    return;
+                }
+                Ok(Ok(_)) => (),
+            }
+
+            drop(client);
+
+            // closing the semaphore causes all the acquire awaits to end
+            ready2.close();
+
+            while let Ok(Some(cmd)) =
+                tokio::time::timeout(config.max_idle, cmd_recv.recv()).await
+            {
+                match cmd {
+                    ConnCmd::SigRecv(sig) => {
+                        use tx5_signal::SignalMessage::*;
+                        #[allow(clippy::single_match)] // placeholder
+                        match sig {
+                            // invalid
+                            HandshakeReq(_) | HandshakeRes(_) => break,
+                            Message(msg) => {
+                                if msg_send.send(msg).await.is_err() {
+                                    break;
+                                }
+                            }
+                            _ => (),
+                        }
+                    }
```

Comment:
Here's the webrtc placeholder. Currently we're just dealing with the `Message` type, but eventually we'll pay attention to offers/answers/ice and try to establish a webrtc connection, switching to it if it is successful.